### PR TITLE
[docker] upgrade vllm version to 0.5.1

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by ydm-amazon — APPROVED** [review]

File: serving/docker/lmi.Dockerfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ ARG auto_gptq_version=0.7.1
 ARG datasets_version=2.19.1
 ARG autoawq_version=0.2.5
 # LMI-Dist Deps
-ARG vllm_wheel="https://publish.djl.ai/vllm/cu124-pt230/vllm-0.4.2%2Bcu124-cp310-cp310-linux_x86_64.whl"
+ARG vllm_wheel="https://github.com/vllm-project/vllm/releases/download/v0.5.1/vllm-0.5.1+cu118-cp310-cp310-manylinux1_x86_64.whl"
```
Comment (lanking520) [inline serving/docker/lmi.Dockerfile:36]: there is a problem here, we should not use the CU118 wheel

**Review by lanking520 — CHANGES_REQUESTED** [review]

File: serving/docker/lmi.Dockerfile (hunk 1 above)
Comment (lanking520) [reply serving/docker/lmi.Dockerfile:36]:
```
https://github.com/vllm-project/vllm/releases/download/v0.5.1/vllm-0.5.1-cp310-cp310-manylinux1_x86_64.whl
```

Comment (sindhuvahinis) [reply serving/docker/lmi.Dockerfile]: Done.

**Comment by lanking520** [comment]
@maaquib will take care on this

**Pull request closed without merging**